src/vs/platform/terminal/common/capabilities/commandDetection/promptInputModel.ts

```diff
@@ -79,6 +82,7 @@ export class PromptInputModel extends Disposable implements IPromptInputModel {
 	private _commandStartX: number = 0;
 	private _lastPromptLine: string | undefined;
 	private _continuationPrompt: string | undefined;
+	private _shellType: TerminalShellType | undefined;
```

Comment:
Could we merge the fish_cancel calls in the SI script into a single one, maybe this?

```
function __vsc_cmd_clear --on-event fish_cancel
	__vsc_esc E "" $__vsc_nonce
	__vsc_esc C
	__vsc_esc D
	set --global _vsc_has_cmd
end
```